base/ocsp/src/main/java/com/netscape/cms/ocsp/LDAPStore.java

```diff
@@ -137,6 +139,7 @@ public void init(IConfigStore config, DBSubsystem dbSubsystem) throws EBaseExcep
                     DEF_CA_CERT_ATTR);
         mByName = mConfig.getBoolean(PROP_BY_NAME, true);
 
+        mCheckConnection = mConfig.getBoolean(PROP_CHECK_SUBSYSTEM_CONNECTION, false);
```

Comment:
This code is only used by the OCSP subsystem. Is there any case where we want this functionality disabled? If it's always going to be enabled maybe we don't need this param.

Reply:
I agree with @edewata, I do not see a reason why this should be disabled so I think we could make permanent this change and not triggered by an option. However, we could just makes it the default behaviour just in case the scenario where disabling the check arise. @ladycfu I am converting to `true`.

I have some concerns with `auths.revocationChecking.enabled=true`. For the OCSP subsystem this work only if the CA sends CRLs through an LDAP and not with direct publishing. Should we stopped, or at least log an error, if it is true but LDAP publishing is not configured?

Reply:
The reason why I propose having a "switch" to go back to the previous behavior was because we want to be sure that there is no gotcha's in the new mechanism.  For now, I think this switch serves its purpose, as a back up .  I think it's fine to have it enabled by default.   Now, wrt auths.revocationChecking.enabled=true, this "Method C" precludes the direct CA->OCSP CRL publishing because the ocsp's ssl port is not able to receive CRL from the CA due to chick-n-egg issue.  The CA depends on the ocsp to start up, and the ocsp depends on the CA to start up, if replying on ca->ocsp direct publishing. The CA is to have enableOCSP=true and its internaldb's  server-cert is to bear an AIA pointing to the ocsp.  @fmarco76 I think you have a good suggestion there on checking whether it's ldapStore that's being set for crl update if ocsp.store.ldapStore.checkSubsystemConnection=true.  btw, what is the concern you wrt auths.revocationChecking.enabled=true ?

Reply:
We need to conduct tests to make sure both ocsp and cap can start in the order of ocsp and then ca.  In the setup, the CA should have its enableOCSP=true in server.xml with both ocspResponderURL and ocspResponderCertNickname removed.  CA's internaldb should have its server-cert bear AIA pointing to the ocsp (instead of the default internal ocsp of the ca).

Reply:
@fmarco76 btw, why do you call this flag  "checkSubsystemConnection"?  To me it seems something like "ocsp.store.ldapStore.validateConnCertWithCRL" would be more telling.  Actually, could you add an explanation in comment where this flag is defined what exactly it does?  We need to add something in the doc.  Thanks.

Reply:
> I think you have a good suggestion there on checking whether it's ldapStore that's being set for crl update if ocsp.store.ldapStore.checkSubsystemConnection=true. 

This setting is done on ldapStore of the OCSP so we do not need to verify if it is used because if it is not then this setting does not apply.

> what is the concern you wrt auths.revocationChecking.enabled=true ?

In the OCSP, if this is true but the ldapStore is not used the revocation cannot be verified. I have not tested if this generate errors. Therefore I was wondering if the user should be notified that revocation will not work even it is enabled.

> why do you call this flag "checkSubsystemConnection"? To me it seems something like "ocsp.store.ldapStore.validateConnCertWithCRL" would be more telling. Actually, could you add an explanation in comment where this flag is defined what exactly it does? We need to add something in the doc. 

Yes, sound better, I will change the name. Where exactly should I write an explanation? A comment in github where the code is provided  or a comment inside the code?

Reply:
wrt auths.revocationChecking.enabled=true, I'm still not sure why it would require ldapStore.  I will take a closer look later.

Reply:
Because the method `OCSPEngine.isRevoked()` is invoked when a new connection to OCSP is created. This method uses the LDAPStore to verify the certificate. If LDAPStore is not used, or the CRL verification is disabled I have not verified what happen.

Reply:
ok I just dug slightly deeper and saw that you had overwritten the isRevoked in the OCSPEngine.